sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionRoundRobin.ts

```diff
@@ -0,0 +1,157 @@
+const serviceBusConnectionString =
+  env["SERVICEBUS_CONNECTION_STRING"] || "<service bus connection string not in environment>";
+
+// NOTE: this sample uses a queue but would also work a session enabled subscription.
+const queueName = env["QUEUE_NAME_WITH_SESSIONS"] || "<queue name not in environment>";
+
+const maxSessionsToProcessSimultaneously = 8;
+const sessionIdleTimeoutSeconds = 3;
+const delayOnErrorMs = 5 * 1000;
+
+// This can be used control when the round-robin processing will terminate
+// by calling abortController.abort().
+const abortController = new AbortController();
+
+// Called just before we start processing the first message of a session.
+// NOTE: This function is used only in the sample and is not part of the Service Bus library.
+async function sessionAccepted(sessionId: string) {
+  console.log(`[${sessionId}] will start processing...`);
+}
+
+// Called by the SessionReceiver when a message is received.
+// This is passed as part of the handlers when calling `SessionReceiver.subscribe()`.
+async function processMessage(msg: ReceivedMessageInfo) {
+  console.log(`[${msg.sessionId}] received message with body ${msg.body}`);
+}
+
+// Called by the SessionReceiver when an error occurs.
+// This will be called in the handlers we pass in `SessionReceiver.subscribe()`
+// and by the sample when we encounter an error opening a session.
+function processError(err: Error, sessionId?: string) {
+  if (sessionId) {
+    console.log(`Error when receiving messages from the session ${sessionId}: `, err);
+  } else {
+    console.log(`Error when creating the receiver for next available session`, err);
+  }
+}
+
+// Called if we are closing a session.
+// `reason` will be:
+// * 'error' if we are closing because of an error(the error will be delivered
+//   to `processError` above)
+// * 'idle_timeout' if `sessionIdleTimeoutMs` milliseconds pass without
+//   any messages being received (ie, session can be considered empty).
+// NOTE: This function is used only in the sample and is not part of the Service Bus library.
+async function sessionClosed(reason: "error" | "idle_timeout", sessionId: string) {
+  console.log(`[${sessionId}] was closed because of ${reason}`);
+}
+
+// utility function to create a timer that can be refreshed
+function createRefreshableTimer(timeoutMs: number, resolve: Function): () => void {
+  let timer: any;
+
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => resolve(), timeoutMs);
+  };
+}
+
+// Queries Service Bus for the next available session and processes it.
+async function receiveFromNextSession(queueClient: QueueClient): Promise<void> {
+  let sessionReceiver = queueClient.createReceiver(ReceiveMode.peekLock, {
+    sessionId: undefined
+  });
+
+  try {
+    await sessionReceiver.getState();
+  } catch (err) {
+    if (
+      (err as MessagingError).condition === "com.microsoft:session-cannot-be-locked" ||
+      (err as MessagingError).condition === "amqp:operation-timeout"
```

Comment:
Can we use the `name` property instead? Its the name of the `MessagingError` that is documented in the public facing apis

Reply:
Absolutely. Swapped over.

The old session manager sample was actually using the ConditionNameMapper which I assume we do NOT want to do. :)